src/select2.component.ts

```diff
@@ -72,7 +72,7 @@ export class Select2Component implements AfterViewInit, OnChanges, OnDestroy, On
         if(changes['value'] && changes['value'].previousValue !== changes['value'].currentValue) {
             const newValue: string = changes['value'].currentValue;
 
-            this.renderer.setElementProperty(this.selector.nativeElement, 'value', newValue);
+            this.setElementValue(newValue);
             this.element.trigger('change.select2');
```

Comment:
move `this.element.trigger('change.select2');` into `setElementValue` function, because it's called every single time as `setElementValue` function is called